### PR TITLE
Auto test space archive status feature, space list support sort, filter by name, status

### PR DESCRIPTION
**Pull request by shuofan**
#### What type of this PR
feature

#### What this PR does / why we need it:
Auto test space archive status feature, space list support sort, filter by name, status

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn @Effet 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |              |
| 🇨🇳 中文    |              |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sfwn — CHANGES_REQUESTED** [review]

File: .erda/migrations/qa/20211029-auto-test-space-archive-status.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE `dice_autotest_space` ADD `archive_status` varchar(191) NOT NULL DEFAULT 'init' COMMENT 'auto test space archive status';
```
[the remark this answers is not in the document]
Comment (shuofan) [reply .erda/migrations/qa/20211029-auto-test-space-archive-status.sql]: done

File: modules/dop/dao/autotest_space.go [hunk 2]
```diff
@@ -50,14 +51,25 @@ func (db *DBClient) CreateAutoTestSpace(space *AutoTestSpace) (*AutoTestSpace, e
...
-func (db *DBClient) ListAutoTestSpaceByProject(projectID int64, pageNo, pageSize int) ([]AutoTestSpace, int, error) {
+func (db *DBClient) ListAutoTestSpaceByProject(req apistructs.AutoTestSpaceListRequest) ([]AutoTestSpace, int, error) {
 	var (
 		space []AutoTestSpace
 		total int
 	)
-	offset := (pageNo - 1) * pageSize
-	if err := db.Where("project_id = ?", projectID).Order("updated_at desc").
-		Offset(offset).Limit(pageSize).Find(&space).
+	offset := (req.PageNo - 1) * req.PageSize
```
[the remark this answers is not in the document]
Comment (shuofan) [reply modules/dop/dao/autotest_space.go:59]: done

File: apistructs/autotest_space.go [hunk 3]
```diff
@@ -34,15 +37,24 @@ var (
 	TestSpaceFailed AutoTestSpaceStatus = "failed"
 )
 
+type AutoTestSpaceArchiveStatus string
+
+const (
+	TestSpaceInit       AutoTestSpaceArchiveStatus = "autoTestSpaceInit"
```
Comment (sfwn) [inline apistructs/autotest_space.go:43]: simplify status in db.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by sfwn** [comment]
Just merge into master, no need to release/1.4.

**Review by sfwn — APPROVED** [review]

**Pull request merged**